src/tir/schedule/primitive/blockize_tensorize.cc

```diff
@@ -565,58 +569,230 @@ StmtSRef Blockize(ScheduleState self, const StmtSRef& loop_sref, bool preserve_u
+            ObjectPtr<VarNode> new_ptr = make_object<VarNode>(*iter_var->var.get());
+            new_ptr->name_hint = "v" + outer_bind->name_hint;
+            auto outer_iter = IterVar(/*dom=*/RangeFromExtent(iter_var->dom->extent),
+                                      /*var=*/Var(new_ptr),
+                                      /*iter_type=*/iter_var->iter_type);
+            if (num_outer_iter_vars == 0) {
+              outer_iter_vars_->push_back(outer_iter);
+              block_var_subst_->Set(iter_var->var, outer_iter->var);
+            }
+          }
+        }
+        ++num_outer_iter_vars;
+        return;
+      }
+    }
+    StmtVisitor::VisitStmt_(block);
+  }
+
+  ScheduleState self_;
+  StmtSRef lca_;
+  Array<IterVar>* outer_iter_vars_;
+  Array<PrimExpr>* outer_bindings_;
+  Array<IterVar>* inner_iter_vars_;
+  Array<PrimExpr>* inner_bindings_;
+  Map<Var, PrimExpr>* block_var_subst_;
+  Array<PrimExpr> out_extent{nullptr};
+  bool in_lca = false;
+  int num_outer_iter_vars = 0;
+  int num_travered = 0;
+};
+
+class BlockizeBlocks : public StmtMutator {
+  Array<StmtSRef> blocks_;
+  StmtSRef lca_;
+  Map<Block, Block>* block_sref_reuse_;
+  BlockRealize* blockized_;
   Array<Stmt> seq_body;
-  PrimExpr outer_predicate{nullptr};
   Array<IterVar> outer_iter_vars{nullptr};
   Array<PrimExpr> outer_bindings{nullptr};
+  Array<IterVar> inner_iter_vars{nullptr};
+  Map<Var, PrimExpr> block_var_subst;
   Array<BufferRegion> read_regions;
   Array<BufferRegion> write_regions;
   std::string outer_block_name = "outer_";
   Map<Var, Var> loop_var_subst;
   arith::Analyzer analyzer;
-  for (const auto& block_sref : block_srefs) {
-    auto block_realize = GetBlockRealize(self, block_sref);
-    auto block = block_realize->block;
-    // Step 1: Derive subspace division
-    std::vector<const ForNode*> loops;
-    Array<Array<arith::IterMark>> division = SubspaceDivide(block_realize, block_sref, lca, &loops,
-                                                            &analyzer, preserve_unit_iters, true);
-    if (division.empty()) {
-      throw SubspaceNotDivisibleError(self->mod, GetRef<For>(loops.back()), block);
-    }
-    outer_predicate = division.back()[0]->extent;
-    PrimExpr inner_predicate = division.back()[1]->extent;
-    // Step 2. Derive block bindings for both outer and inner block.
-    Array<IterVar> inner_iter_vars;
-    Array<PrimExpr> inner_bindings;
-    Map<Var, PrimExpr> block_var_subst =                       //
-        DeriveBlockBinding(block->iter_vars, division,         //
-                           &outer_iter_vars, &outer_bindings,  //
-                           &inner_iter_vars, &inner_bindings,  //
-                           preserve_unit_iters, outer_iter_vars.defined());
-    // Step 3: Do var substitution to adjust to the new block bindings
+  Block tmp_in_block;
+  Map<Var, arith::IntSet> inner_iter_dom;
+  bool _first_in = false;
```

Comment:
It would be great to make variable name consistent for `_first_in`, `target_in`, `IsBlockNode`

Reply:
done!